FhirToDataLake/SynapseValidate/Synapse_Validator.py

```diff
@@ -0,0 +1,207 @@
+    Queried = "queried"
+    FieldDict = "field_mapping_dict"
+
+    def __init__(self, sql_client, expected_data_client, field_mapping_dicts):
+        self.sql_client = sql_client
+        self.expected_data_client = expected_data_client
+        self.field_mapping_dicts = field_mapping_dicts
+
+    def fetch(self, resource_type):
+        result = {}
+        result[DataFactory.ResourceType] = resource_type
+        result[DataFactory.Queried] = self.sql_client.get_data(resource_type)
+        result[DataFactory.Expected] = self.expected_data_client.get_all_entries(resource_type)
+        result[DataFactory.FieldDict] = self.field_mapping_dicts[resource_type]
+        return result
+
+
+if __name__ == "__main__":
+    sys.stdout.flush()
+
+    print('-> Executing script file: ', str(sys.argv[0]))
+    print('-> SynapseServerEndpoint: ', Sql_Server_Endpoint)
+    print('-> Database: ', Database)
+    print('-> FhirServerUrl: ', Fhir_Server_Base_Url)
+
+    schema_manager = SchemaManager(Schema_Collection_Directory)
+    resource_types = schema_manager.get_all_resource_types()
+
+    data_source = RawDataSource(Fhir_Server_Base_Url)
+    sql_client = SQLServerClient(Sql_Server_Endpoint, Database, Sql_Username, Sql_Password)
+
+    dataFactory = DataFactory(sql_client, data_source, schema_manager.generate_field_mapping_dicts())
+
+    start_time = time.time()
+    with concurrent.futures.ThreadPoolExecutor(max_workers=10) as executor:
+        futures = [executor.submit(dataFactory.fetch, resource_type) for resource_type in resource_types]
+        for future in concurrent.futures.as_completed(futures):
+            result = future.result()
+            resource_type = result[DataFactory.ResourceType]
+            expected_entries = result[DataFactory.Expected]
+            queried_df = result[DataFactory.Queried]
+            field_mapping_dict = result[DataFactory.FieldDict]
+            print(f"Get {queried_df.shape[0]} resources from \"{resource_type}\", columns number is {queried_df.shape[1]}")
+
+            if len(expected_entries) != queried_df.shape[0]:
+                raise Exception(f"Resource \"{resource_type}\" rows number is incorrect. "
+                                f"Expected resource number is {len(expected_entries)}, "
+                                f"while rows number on Synapse is {queried_df.shape[0]}.")
+
+            if len(field_mapping_dict) != queried_df.shape[1]:
+                expected_columns = set(field_mapping_dict.keys())
+                queried_columns = set(queried_df.columns.tolist())
+                raise Exception(f"Resource \"{resource_type}\" columns number is incorrect. "
+                                f"Expected column number is {len(field_mapping_dict)}, "
+                                f"while column number number on Synapse is {queried_df.shape[1]}.")
+
+    print(f"Validate {len(resource_types)} resource types complete in " + str(time.time() - start_time), "seconds")
```

Comment:
Nit: Typo `Validating .... completes in ...`

Reply:
Fixed, thanks.